R/tmiic.utils.R

```diff
@@ -0,0 +1,1343 @@
+  #
+  if ( ! ("delta_t" %in% colnames(state_order)) )
+    {
+    # Add delta_t column with 0 for contextual, NA otherwise
+    #
+    state_order$delta_t = NA
+    state_order$delta_t[ state_order$is_contextual == 1] = 0
+    }
+  else
+    {
+    # Replace NA vals if possible:
+    #
+    na_in_so = is.na (state_order$delta_t)
+    are_contextual = (state_order$is_contextual == 1)
+    if ( any (na_in_so & are_contextual) )
+      {
+      # For contextual, replace NAs with 1
+      #
+      msg_str = list_to_str (state_order$var_names[na_in_so & are_contextual],
+                             n_max=10)
+      miic_warning ("temporal checks", "the missing delta t have been",
+        " set to 0 for contextual variables (", msg_str, ").")
+      state_order$delta_t[na_in_so & are_contextual] = 0
+      }
+    #
+    # Contextual vars done, look if still NAs on not contextual
+    #
+    na_in_so = is.na (state_order$delta_t)
+    if ( any (na_in_so) )
+      {
+      # For non contextual vars with delta_t equal to NA:
+      # - if no other var has a delta_t, go for automatic estimate
+      # - if there is an unique delta_t, apply this value to all
+      # - if there is multiple delta_t values, stop
+      #
+      uniq_vals = unique (state_order$delta_t[(!na_in_so) & (!are_contextual)])
+      if (length (uniq_vals) == 0)
+        {
+        msg_str = list_to_str (state_order$var_names[na_in_so], n_max=10)
+        miic_warning ("state order", "the missing delta t will be ",
+          " determined from data for variables ", msg_str, ".")
+        }
+      else if (length (uniq_vals) > 1)
+        {
+        miic_error ("state_order", "the state order contains NAs",
+          " for the delta t and it can not be completed",
+          " automatically as multiple values are already present.")
+        }
+      else
+        {
+        msg_str = list_to_str (state_order$var_names[na_in_so], n_max=10)
+        miic_warning ("state order", "the missing delta t will be ",
+          " set to ", uniq_vals, " for variables ", msg_str, ").")
+        state_order$delta_t[na_in_so & are_contextual] = uniq_vals
+        }
+      }
+    #
+    # Check/fix invalid values: for contextual vars, delta_t must be 0
+    #
+    wrongs = ( (!is.na (state_order$delta_t))
+             & (state_order$delta_t != 0)
+             & (state_order$is_contextual == 1) )
+    if ( any (wrongs) )
+      {
+      msg_str = list_to_str (state_order$var_names[wrongs], n_max=10)
+      if (sum (wrongs) == 1)
+        miic_warning ("temporal checks", "the variable ", msg_str, ", as",
+          " contextual, has an invalid delta t. It will be set to 0.")
+      else
+        miic_warning ("temporal checks", "several variables (", msg_str, "), as",
+          " contextual, have an invalid delta_t. They will be set to 0.")
+      state_order$delta_t[wrongs] = 0
+      }
+    #
+    # Warning if multiple values of delta_t excluding contextual
```

Comment:
EN (OD)

Reply:
Warning if multiple values of delta_t (excluding contextual) are present